pilot/pkg/serviceregistry/serviceregistry_test.go

```diff
@@ -824,7 +824,6 @@ func TestWorkloadInstances(t *testing.T) {
 
 		newSvc.Spec.Ports[0].TargetPort = intstr.IntOrString{IntVal: 9090}
 		makeService(t, s.KubeClient().Kube(), newSvc)
-		expectEndpoints(t, s, "outbound|80||service.namespace.svc.cluster.local", nil, nil)
```

Comment:
I donot think this fix can totally eliminate data race

It seems we need to lock shard as well in `AssertEndpointConsistency`

Reply:
The logic of `UntilSuccessOrFail` within method `expectEndpoints` has alreay ensured that the endpointShard has been updated. So the concurrency problem will not happen in `AssertEndpointConsistency`.

Reply:
A rough look, it is here
```
WARNING: DATA RACE
Read at 0x00c0011ef380 by goroutine 1301:
  istio.io/istio/pilot/pkg/xds.(*FakeDiscoveryServer).AssertEndpointConsistency()
      /home/prow/go/src/istio.io/istio/pilot/pkg/xds/fake.go:520 +0xa8b
  istio.io/istio/pilot/pkg/serviceregistry_test.expectEndpoints()
      /home/prow/go/src/istio.io/istio/pilot/pkg/serviceregistry/serviceregistry_test.go:1206 +0x1d2
  istio.io/istio/pilot/pkg/serviceregistry_test.TestWorkloadInstances.func23()
      /home/prow/go/src/istio.io/istio/pilot/pkg/serviceregistry/serviceregistry_test.go:827 +0x4bb
```

Reply:
Yes. It is becase the svc `outbound|80||service.namespace.svc.cluster.local` has already been deleted in last check. If we check again, it will return from UntilSuccessOrFail immediately and begin to do `AssertEndpointConsistency`.

But the operation of latest `makeService` is still processing which will data race with the following `AssertEndpointConsistency`.

Reply:
This is removing test coverage though